### PR TITLE
Reorg the builders, precompiles, and pallets sections

### PR DESCRIPTION
**Pull request by eshaben**
### Description

This PR reorganizes the builders section into a higher-level Ethereum / Substrate separation and categorizes the precompiles and pallets sections. Relabeling pallets as low-level interfaces for immediate understanding of what pallets offer.

Goes with mkdocs PR: https://github.com/papermoonio/moonbeam-mkdocs/pull/195

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] I have run my changes through Grammarly
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interoperability/xcm/xcm-sdk/v1/xcm-sdk.md [hunk 1]
```diff
@@ -215,7 +215,7 @@ If you want to pass in a browser extension wallet to viem, you can use the follo
 
 ### Create a Polkadot Signer {: #create-a-polkadot-signer }
 
-In this example, you can use a [Polkadot.js Keyring](/builders/build/substrate-api/polkadot-js-api#keyrings){target=\_blank} to sign transactions. Please note that this approach is not recommended for production applications.
+In this example, you can use a [Polkadot.js Keyring](/builders/substrate/polkadot-js-api#keyrings){target=\_blank} to sign transactions. Please note that this approach is not recommended for production applications.
```
Comment (themacexpert) [inline builders/interoperability/xcm/xcm-sdk/v1/xcm-sdk.md:218]:
```suggestion
In this example, you can use a [Polkadot.js Keyring](/builders/substrate/libraries/polkadot-js-api#keyrings){target=\_blank} to sign transactions. Please note that this approach is not recommended for production applications.
```

File: builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-precompile.md [hunk 2]
```diff
@@ -293,13 +293,13 @@ For this example, you'll interact with the `transactThroughSigned` function of t
         const feeLocationAddress = '0xFFFFFFFF1AB2B146C526D4154905FF12E6E57675';
         ```
 
-    - `transactRequiredWeightAtMost` - the weight required to execute the call in the `Transact` instruction. You can get this information by using the [`paymentInfo` method of the Polkadot.js API](/builders/build/substrate-api/polkadot-js-api#fees){target=\_blank} on the call
+    - `transactRequiredWeightAtMost` - the weight required to execute the call in the `Transact` instruction. You can get this information by using the [`paymentInfo` method of the Polkadot.js API](/builders/substrate/polkadot-js-api#fees){target=\_blank} on the call
```
Comment (themacexpert) [inline builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-precompile.md:296]:
```suggestion
    - `transactRequiredWeightAtMost` - the weight required to execute the call in the `Transact` instruction. You can get this information by using the [`paymentInfo` method of the Polkadot.js API](/builders/substrate/polkadot-js-api#fees){target=\_blank} on the call
```
```suggestion
    - `transactRequiredWeightAtMost` - the weight required to execute the call in the `Transact` instruction. You can get this information by using the [`paymentInfo` method of the Polkadot.js API](/builders/substrate/libraries/polkadot-js-api#fees){target=\_blank} on the call
```

File: builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-pallet.md [hunk 3]
```diff
@@ -369,7 +369,7 @@ Since you'll be interacting with the `transactThroughSigned` function of the XCM
     ```
 
     !!! note
-        For accurate estimates of the `refTime` and `proofSize` figures for `transactRequiredAtMost`, you can use the [`paymentInfo` method of the Polkadot.js API](/builders/build/substrate-api/polkadot-js-api#fees){target=\_blank}.
+        For accurate estimates of the `refTime` and `proofSize` figures for `transactRequiredAtMost`, you can use the [`paymentInfo` method of the Polkadot.js API](/builders/substrate/polkadot-js-api#fees){target=\_blank}.
```
Comment (themacexpert) [inline builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-pallet.md:372]:
```suggestion
        For accurate estimates of the `refTime` and `proofSize` figures for `transactRequiredAtMost`, you can use the [`paymentInfo` method of the Polkadot.js API](/builders/substrate/libraries/polkadot-js-api#fees){target=\_blank}.
```

File: builders/ethereum/precompiles/features/governance/preimage.md [hunk 4]
```diff
@@ -98,14 +98,14 @@ Then take the following steps:
  3. Enter the text of the remark, ensuring it is unique. Duplicate proposals such as "Hello World!" will not be accepted
  4. Click the **Submit preimage** button but don't sign or confirm the transaction on the next page 
 
-![Get the proposal hash](/images/builders/pallets-precompiles/precompiles/democracy/democracy-5.webp)
+![Get the proposal hash](/images/builders/ethereum/precompiles/democracy/democracy-5.webp)
```
Comment (themacexpert) [inline builders/ethereum/precompiles/features/governance/preimage.md:101]: This image is missing

File: builders/ethereum/precompiles/features/governance/preimage.md [hunk 5]
```diff
@@ -98,14 +98,14 @@ Then take the following steps:
...
 
 On the next screen, take the following steps:
 
  1. Press the triangle icon to reveal the encoded proposal in bytes
  2. Copy the **bytes** representing the encoded proposal - you'll need this when calling the `notePreimage` function in a later step
 
-![Get the encoded proposal](/images/builders/pallets-precompiles/precompiles/democracy/democracy-6.webp)
+![Get the encoded proposal](/images/builders/ethereum/precompiles/democracy/democracy-6.webp)
```
Comment (themacexpert) [inline builders/ethereum/precompiles/features/governance/preimage.md:108]: This image is missing

File: builders/ethereum/precompiles/features/governance/preimage.md [hunk 6]
```diff
@@ -89,7 +89,7 @@ In this section, you'll get the preimage hash and the encoded proposal data for
  2. Select **Preimages** from the dropdown
  3. From the **Preimages** page, click on **+ Add preimage**
 
-![Add a new preimage](/images/builders/pallets-precompiles/precompiles/democracy/democracy-4.webp)
+![Add a new preimage](/images/builders/ethereum/precompiles/democracy/democracy-4.webp)
```
Comment (themacexpert) [inline builders/ethereum/precompiles/features/governance/preimage.md:92]: This image is missing

File: builders/ethereum/libraries/ethersrs.md [hunk 7]
```diff
@@ -606,8 +606,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
 If successful, the transaction receipt will be displayed in the terminal. You can use the `read_number` function in the `main` function to make sure that value is changing as expected. If you're using the `read_number` function after resetting the number, you should see `0` printed to the terminal.
 
-![Terminal logs from resetting the number](/images/builders/build/eth-api/libraries/ethersrs/ethersrs-4.webp)
+![Terminal logs from resetting the number](/images/builders/ethereum/libraries/ethersrs/ethersrs-4.webp)
 
-You can view the [complete script on GitHub](https://raw.githubusercontent.com/moonbeam-foundation/moonbeam-docs/master/.snippets/code/builders/build/eth-api/libraries/ethers-rust/deploy-contract/main.rs){target=\_blank}.
+You can view the [complete script on GitHub](https://raw.githubusercontent.com/moonbeam-foundation/moonbeam-docs/master/.snippets/code/builders/ethereum/libraries/ethers-rust/deploy-contract/main.rs){target=\_blank}.
```
Comment (themacexpert) [inline builders/ethereum/libraries/ethersrs.md:611]: This link is broken for me but maybe it will be all set once this is merged into master.

**Review by themacexpert — CHANGES_REQUESTED** [review]
Just a few broken links everything looks great otherwise!

**Review by albertov19 — APPROVED** [review]
Amazing job! I'm super sold on the new structure! Love it!

We should replicate this for Tanssi when possible because I think it is very intuitive!

**Comment by eshaben** [comment]
@themacexpert thank you for such thorough reviews!! Love it!!

**Review by themacexpert — APPROVED** [review]
LGTM 👍

**Pull request merged**